app/src/main/java/net/gsantner/opoc/util/GsFileUtils.java

```diff
@@ -752,4 +754,40 @@ public static List<File> replaceFilesWithCachedVariants(@Nullable List<File> fil
         }
         return files;
     }
+
+    // Check if a file can be created (parent exists and can be written)
+    public static boolean isWritable(final File file) {
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
+            return file != null && Files.isWritable(file.toPath());
+        } else{
+            return file != null && file.canWrite();
+        }
+    }
+
+    // Check if a file can be created (parent exists and can be written)
+    public static boolean canCreate(final File file) {
+        return isWritable(file) || isWritable(file.getParentFile());
+    }
+
+    // Returns true if test is a child of parent. A file is not a child of itself
+    public static boolean isChild(final File parent, File test) {
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
+            return !parent.equals(test) && test.toPath().toAbsolutePath().startsWith(parent.toPath().toAbsolutePath());
+        } else {
+            // Can fail is parent dest name contains test. Not a good method
+            return !parent.equals(test) && test.getAbsolutePath().startsWith(parent.getAbsolutePath());
+        }
+    }
+
+    public static File join(final File ... files) {
```

Comment:
more intentrevealing: rename to joinFilePaths

Reply:
It's in `GsFileUtils`. IMO the namespace gives us the necessary context. For reference, python's built in equivalent is `os.path.join`

Reply:
Just join is fine here :-)